### PR TITLE
replace by fee handling for deposits

### PR DESCRIPTION
**Pull request by hozzjss**
This checks if the deposit tx exists and if it doesn't it fetches if it has been rbf'ed and if so redirect to the rbf'ed tx after notifying emily

### CODE REVIEW

File: src/actions/bitcoinClient.ts [hunk 1]
```diff
@@ -80,8 +80,16 @@ export const scanTxOutSet = async (
...
+): Promise<BitcoinTransactionResponse | null> => {
   const result = await fetch(`${env.MEMPOOL_API_URL}/tx/${txid}`);
+  if (result.status === 404) {
+    return null;
+  }
+  return await result.json();
+};
+
+export const getTxRbf = async (txid: string): Promise<boolean> => {
+  const result = await fetch(`${env.MEMPOOL_API_URL}/v1/tx/${txid}/rbf`);
```
Comment (matteojug) [inline src/actions/bitcoinClient.ts:92]: We don't have the `v1` part above, is it expected?

Comment (matteojug) [inline src/actions/bitcoinClient.ts:92]: Should this also have the `result.status` guard?

**Review by matteojug — APPROVED** [review]

File: src/actions/bitcoinClient.ts (hunk 1 above)
Comment (hozzjss) [reply src/actions/bitcoinClient.ts:92]: yes for this one it is

Comment (hozzjss) [reply src/actions/bitcoinClient.ts:92]: actually no, and I'm being optimistic here tbh, basically the user can do rbf which could cancel the transaction which breaks everything, that case is not handled

**Pull request merged**